src/runtime-rs/crates/resource/src/network/network_pair.rs

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) 2019-2022 Alibaba Cloud
+// Copyright (c) 2019-2022 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::{convert::TryFrom, sync::Arc, usize};
+
+use anyhow::{anyhow, Context, Result};
+use futures::stream::TryStreamExt;
+
+use super::{
+    network_model,
+    utils::{self, address::Address, link},
+};
+
+#[derive(Default, Copy, Clone, Debug, PartialEq, Eq)]
+pub struct NetInterworkingModel(u32);
+
+#[derive(Default, Debug, Clone)]
+pub struct NetworkInterface {
+    pub name: String,
+    pub hard_addr: String,
+    pub addrs: Vec<Address>,
+}
+
+#[derive(Default, Debug)]
+pub struct TapInterface {
+    pub id: String,
+    pub name: String,
+    pub tap_iface: NetworkInterface,
+}
+#[derive(Debug)]
+pub struct NetworkPair {
+    pub tap: TapInterface,
+    pub virt_iface: NetworkInterface,
+    pub model: Arc<dyn network_model::NetworkModel>,
+    pub network_qos: bool,
+    pub ant_dns_filter: bool,
```

Comment:
ant?

Reply:
Thanks for your review, I think need to remove this.